src/main/java/com/epam/emotionalhelp/service/impl/QuizResultServiceImpl.java

```diff
@@ -1,22 +1,172 @@
+            quiz.ifPresent(quizResult::setQuiz);
+            quizResult.setAttempt(quizAttemptFinal);
+            quizResult.setEmotion(entry.getKey());
+            quizResult.setScore(entry.getValue());
+            quizResultRepository.save(quizResult);
+        }
+        return new AttemptDto(quizAttempt.getId());
+    }
+
+    @Override
+    public List<EmotionalMapDto> findQuizResultsByAttemptId(Long attemptId) {
+        //Find QuizResult(Emotion , Overall score)
+        var list = quizResultRepository.findAllByAttemptId(attemptId);
+
+        //Create list to return
+        var resultList = new ArrayList<EmotionalMapDto>();
+        //Extract Emotion and get Subcategories based on '%'
+        for (QuizResult quizResult : list) {
+            var emotion = quizResult.getEmotion();
+            //Create QuizResultEmotionDto
+            var quizResultEmotionDto = new EmotionalMapDto();
+            quizResultEmotionDto.setCategory(emotion.getDescription());
+            //Find All Subcategories
+            //If score is equal to 0 it means that list with subcategories should be empty
+            if (quizResult.getScore() == 0) {
+                quizResultEmotionDto.setSubCategories(new ArrayList<>());
+            } else {
+                int percentage = (int) (((double) quizResult.getScore() / MAX_SCORE_VALUE) * PERCENTAGE_VALUE);
+                var allSubcategories = subcategoryRepository.findAllSubcategories(percentage, emotion.getId());
+                var subcategories = allSubcategories
+                        .stream()
+                        .limit(SUBCATEGORIES_LIMIT_VALUE)
+                        .sorted(Comparator.comparing(Subcategory::getWeight))
+                        .collect(Collectors.toList());
+                var subcategoryList = ListUtils.partition(subcategories, MAX_CONTAINER_SIZE);
+                var categoryList = new ArrayList<SubcategoryContainerDto>();
+                for (List<Subcategory> partition : subcategoryList) {
+                    var category = partition.stream().map(Subcategory::getDescription).collect(Collectors.toSet());
+                    double score = (double) partition.stream().mapToInt(Subcategory::getWeight).max().orElse(0) / PERCENTAGE_VALUE;
+                    categoryList.add(new SubcategoryContainerDto(category, score));
+                }
+                quizResultEmotionDto.setSubCategories(categoryList);
+            }
+            //Add QuizResultEmotionDto to the List
+            resultList.add(quizResultEmotionDto);
+        }
+        return resultList;
+    }
+
+    private Map<EmotionCategory, List<EmotionDto>> filterListByEmotions(List<EmotionDto> inputEmotions) {
+        Map<EmotionCategory, List<EmotionDto>> map = new HashMap<>();
+        List<EmotionDto> emotions;
+        if (inputEmotions.size() > LIST_SLICE_SIZE) {
+            emotions = new ArrayList<>(inputEmotions.subList(0, LIST_SLICE_SIZE));
```

Comment:
Simplify to
 ArrayList<>(inputEmotions.subList(0, LIST_SLICE_SIZE));
If list size is 6 you slice anyway 0 to 6